File: FoodToOrderFrontend/src/app/models/restaurant.ts
import { Address } from './address';
import { Dishes } from './dishes';
export class Resturants {
  id: string;
  rName: string;
  user_id: string;
  dishlist: Dishes[];
  arrAddress: Address[];
  location: string = '';
  ROpen: boolean;
  Type:string

  constructor(
    id: string,
    n: string,
    uid: string,
    d: Dishes[],
    arrAddr: Address[],
    open: boolean,
    type:string
  ) {
    this.id = id;
    this.rName = n;
    this.user_id = uid;
    this.dishlist = d;
    this.arrAddress = arrAddr;
    this.ROpen = open;
    this.Type = type
  }
}

File: FoodToOrderBackend/FoodToOrderApi/FoodToOrderDB/Restaurant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace FoodToOrderDB;


[Table("Restaurant")]
public partial class Restaurant
{
    [Key]
    public int id { get; set; }

   
    public string rName { get; set; } 

 
    public bool ROpen { get; set; }

    public int? user_id { get; set; }
    public string location { get; set; }

    public string Type { get; set; }
    public ICollection<Dish>? dishlist { get; set; }
    public ICollection<Address>? arrAddress { get; set; }
}

 